ext/vulnsrc/redhat/redhat.go

```diff
@@ -0,0 +1,208 @@
+}
+
+type NVRA struct {
+	Name    string
+	Version string
+	Release string
+	Arch    string
+}
+
+type VmaasPostRequest struct {
+	ErrataList    []string `json:"errata_list"`
+	ModifiedSince string   `json:"modified_since"`
+}
+
+type updater struct{}
+
+func init() {
+	vulnsrc.RegisterUpdater("redhat", &updater{})
+}
+
+func (u *updater) Update(datastore database.Datastore) (resp vulnsrc.UpdateResponse, err error) {
+	log.WithField("package", "RedHat").Info("Start fetching vulnerabilities")
+
+	// Get the timestamp from last scan
+	flagValue, ok, err := database.FindKeyValueAndRollback(datastore, updaterFlag)
+	if err != nil {
+		return resp, err
+	}
+	timeNow := time.Now()
+	newTime := timeNow.Format(time.RFC3339)
+	var rhsaSince string
+	if !ok {
+		rhsaSince = rhsaFirstTime
+	} else {
+		rhsaSince = flagValue
+	}
+
+	requestParames := VmaasPostRequest{
+		ErrataList:    []string{"RHSA-.*"},
+		ModifiedSince: rhsaSince,
+	}
+
+	// Fetch the update list.
+	advisoriesURL := vmaasURL + "/errata/"
+	r, err := httputil.PostWithUserAgent(advisoriesURL, requestParames)
+	if err != nil {
+		log.WithError(err).Error("could not download RedHat's update list")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+	defer r.Body.Close()
+
+	if !httputil.Status2xx(r) {
+		log.WithField("StatusCode", r.StatusCode).Error("Failed to update RedHat")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+
+	var rhsaData RHSAdata
+	if err := json.NewDecoder(r.Body).Decode(&rhsaData); err != nil {
+		return resp, err
+	}
+
+	for _, advisory := range rhsaData.ErrataList {
+		vulnerabilities := parseAdvisory(advisory)
+		resp.Vulnerabilities = append(resp.Vulnerabilities, vulnerabilities...)
+
+	}
+	if len(resp.Vulnerabilities) > 0 {
+		log.WithFields(log.Fields{
+			"items":   len(resp.Vulnerabilities),
+			"updater": "RedHat",
+		}).Debug("Found new vulnerabilities")
+	}
+
+	// save new timestamp to database
+	resp.FlagName = updaterFlag
+	resp.FlagValue = newTime
+	return resp, nil
+
+}
+
+// parseAdvisory - parse advisory metadata and create new Vulnerabilities objects
+func parseAdvisory(advisory Advisory) (vulnerabilities []database.VulnerabilityWithAffected) {
+	for _, cve := range advisory.CveList {
+		vulnerability := database.VulnerabilityWithAffected{
+			Vulnerability: database.Vulnerability{
+				Name:        cve,
+				Link:        cveURL + cve,
+				Severity:    severity(advisory.Severity),
+				Description: advisory.Description,
+			},
+		}
+		for _, nvra := range advisory.PackageList {
+			parsedNVRA := parseNVRA(nvra)
+			p := database.AffectedFeature{
+				FeatureName:     parsedNVRA.Name,
+				AffectedVersion: parsedNVRA.Version + "-" + parsedNVRA.Release,
+				FixedInVersion:  parsedNVRA.Version + "-" + parsedNVRA.Release,
+				AffectedType:    affectedType,
+				Namespace: database.Namespace{
+					// Not sure what namespace should I use here (can we use CPE here?)
+					// TODO: fix it
+					Name:          "redhat",
```

Comment:
Namespace is still not clear and has to be fixed. Any advice what to use here?

Reply:
namespace in this case should be the name of the operating system plus the version of the operating system
e.g. 
https://github.com/coreos/clair/blob/5cd6a8cc921be3fc836c9b90d5f0b0bf30d753ac/ext/vulnsrc/rhel/rhel.go#L353

Reply:
The previous RH security data (Oval) data contains only advisories for rhel product, but the vmaas contains whole Red Hat's portfolio (rhel, rhscl, rhos, rhose and many more), so I am not sure what you mean by the version of operating system.

Is it documented what is the namespace used for in clair?